UnityProject/Assets/Scripts/Items/Food/Edible.cs

```diff
@@ -200,29 +206,34 @@ public ReagentMix GetMixForBite(PlayerScript feeder)
 			{
 
 				currentBites--;
-				if (leavings != null)
-				{
-					var leavingsInstance = Spawn.ServerPrefab(leavings).GameObject;
-					var pickupable = leavingsInstance.GetComponent<Pickupable>();
-					bool added = false;
-					var ToDropOn = gameObject;
 
-					if (feeder != null)
-					{
-						var feederSlot = feeder.DynamicItemStorage.GetActiveHandSlot();
-
-						ToDropOn = feeder.gameObject;
-						added = Inventory.ServerAdd(pickupable, feederSlot);
-					}
 
-					if (added == false)
+				if (currentBites <= 0)
+				{
+					if (leavings != null)
 					{
-						//If stackable has leavings and they couldn't go in the same slot, they should be dropped
-						pickupable.UniversalObjectPhysics.DropAtAndInheritMomentum(
-							ToDropOn.GetComponent<UniversalObjectPhysics>());
+						var leavingsInstance = Spawn.ServerPrefab(leavings).GameObject;
+						var pickupable = leavingsInstance.GetComponent<Pickupable>();
+						bool added = false;
+						var ToDropOn = gameObject;
+
+						if (feeder != null)
+						{
+							var feederSlot = feeder.DynamicItemStorage.GetActiveHandSlot();
+
+							ToDropOn = feeder.gameObject;
+							added = Inventory.ServerAdd(pickupable, feederSlot, ReplacementStrategy.DropOther);
+						}
+
+						if (added == false)
+						{
+							//If stackable has leavings and they couldn't go in the same slot, they should be dropped
+							pickupable.UniversalObjectPhysics.DropAtAndInheritMomentum(
+								ToDropOn.GetComponent<UniversalObjectPhysics>());
+						}
 					}
+					_ = Inventory.ServerDespawn(gameObject);
```

Comment:
this looks like it breaks my own fix

Reply:
tthis only triggers the ServerDespawn when  if (currentBites <= 0)

Reply:
...which is the intended behavior?

Reply:
yeah, so it's fine?

Reply:
I had to re-read this twice to make sure I'm not tripping.
but there's nothing preventing this from despawning the item before currentBites reaches zero, so you're re-introducing a bug that I already fixed.